bpf/bpf_netdev.c

```diff
@@ -443,9 +444,33 @@ static __always_inline int do_netdev(struct __sk_buff *skb, __u16 proto)
 			if (l3_csum_replace(skb, ETH_HLEN + offsetof(struct iphdr, check),
 			    0, sum, 0) < 0)
 				return DROP_CSUM_L3;
+
+#ifdef HAVE_FIB_LOOKUP
+			{
+			int err;
+
+			if (!revalidate_data(skb, &data, &data_end, &iphdr))
+				return DROP_INVALID;
+
+			fib_params.family = AF_INET;
+			fib_params.ifindex = ENCRYPT_IFACE;
+
+			fib_params.ipv4_src = iphdr->saddr;
+			fib_params.ipv4_dst = iphdr->daddr;
+
+			err = fib_lookup(skb, &fib_params, sizeof(fib_params),
+					    BPF_FIB_LOOKUP_DIRECT | BPF_FIB_LOOKUP_OUTPUT);
+			if (err != 0)
+				return DROP_INVALID;
+			if (eth_store_daddr(skb, fib_params.dmac, 0) < 0)
+				return DROP_WRITE_ERROR;
+			if (eth_store_saddr(skb, fib_params.smac, 0) < 0)
+				return DROP_WRITE_ERROR;
+			}
```

Comment:
nit: Whitespace seems a bit weird here.

Reply:
yeah so added the braces to add the `int err` and avoid having another ifdef/endif at the top of the function. But didn't want to indent the lines even more. I don't really care much but because they the braces don't signify a if/switch/etc case I don't mind them leaving them this way.